lib/facter/self_service.rb

```diff
@@ -188,4 +188,36 @@
     # Is puppet_metrics_collector::system configured
     { S0040: PuppetSelfService.service_running_enabled('puppet_system_processes-metrics.timer') }
   end
+
+  chunk(:S0027) do
+    next unless PuppetSelfService.primary?
+    # Check if thundering herd is occuring.
+    psql_output_data = PuppetSelfService.psql_thundering_herd
+    if psql_output_data.nil? || psql_output_data.empty?
+      { S0027: true }
+    else
+      thundering_query_output = psql_output_data.split(' ').drop(2).map(&:to_i)
+      thundering_query_output.delete(0)
+      if thundering_query_output.max.to_i < 20 || thundering_query_output.min.to_i < 1
+        # not meeting minimum requirement
+        { S0027: true }
+      else
+        # minvalue = thundering_query_output.min.to_i
+        # maxvalue = thundering_query_output.max.to_i
+        # differences = ((maxvalue - minvalue) * 100) / minvalue
+        # { S0027: differences <= 75 }
+
+        # Changing the approach based on standard deviation method
+        # taking only the first n number of items to check if thundering herd is occured or not.
```

Comment:
We need to clean up the comments and older code iterations from the PR
Comments should be relevant to the end user or future developer